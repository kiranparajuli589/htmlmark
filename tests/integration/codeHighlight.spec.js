import HtmlMark from "../../lib/index.js"
import hljs from "highlight.js"


const mdp = new HtmlMark({
	highlightFn: (code, lang) => {
		if (lang && hljs.getLanguage(lang)) {
			return hljs.highlight(code, { language: lang }).value
		} else {
			return hljs.highlightAuto(code).value
		}
	}
})

describe("code highlight support", () => {
	test("should highlight code blocks", () => {
		const str = `
\`\`\`javascript
const a = 1
\`\`\`
`
		expect(mdp.parse(str)).toMatchSnapshot()
	})
})
