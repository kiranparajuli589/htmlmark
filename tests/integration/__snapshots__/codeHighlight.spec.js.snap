// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`code highlight support should highlight code blocks 1`] = `
"<pre><code class='language-javascript'><span class=\\"hljs-keyword\\">const</span> a = <span class=\\"hljs-number\\">1</span></code></pre>
"
`;
